src/main/cpp/src/cast_string_to_datetime.cu

```diff
@@ -1113,17 +1112,29 @@ std::unique_ptr<cudf::column> parse_timestamp_strings(cudf::strings_column_view
                                                       int64_t default_epoch_day,
                                                       cudf::column_view const& tz_info,
                                                       cudf::table_view const& transitions,
-                                                      bool is_spark_320,
+                                                      int platform_ordinal,
+                                                      int major_version,
+                                                      int minor_version,
+                                                      int fix_version,
                                                       rmm::cuda_stream_view stream,
                                                       rmm::device_async_resource_ref mr)
 {
+  platform_type const platform = spark_rapids_jni::get_platform_type(platform_ordinal);
+  int v                        = major_version * 100 + minor_version * 10 + fix_version;
+  bool is_spark_320            = (platform == platform_type::SPARK && v == 320);
+  bool is_spark_400_or_later   = (platform == platform_type::SPARK && v >= 400);
+  int v_for_db                 = major_version * 100 + minor_version;
```

Comment:
Why for `v_for_db`, `minor_version` is not multiply by `10`?

Reply:
For DB, its major version is 2 digits, so it has little probablity that minor and fix versions are also 2 digits.
So here use multiple by 100, it's similar to multiple by 10.

Reply:
Alright, I think we should better store all the major and minor/patch version numbers. Then when we query, we can do like this:
```
struct spark_system {
....
private:
 int major, minor, patch;
};

auto const is_spark_320 = sys.platform() == platform_type::XXX && sys.major() == 3 && sys.minor() = 2 && sys.patch() == 0;

auto const is_db_14_3_or_later = sys.platform() == platform_type::DATABRICKS && sys.major() == 14 && sys.
minor() = 3;

etc.
```